src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/TranscriptionDownloader.java

```diff
@@ -44,37 +41,36 @@ public DownloadTranscriptResponse downloadTranscript(Integer transcriptionId) {
             .max(comparing(TranscriptionDocumentEntity::getUploadedDateTime))
             .orElseThrow(() -> new DartsApiException(FAILED_TO_DOWNLOAD_TRANSCRIPT));
 
-        var latestExternalObjectDirectory = latestTranscriptionDocument.getExternalObjectDirectoryEntities()
-            .stream()
-            .filter(dir -> nonNull(dir.getExternalLocation()))
-            .max(comparing(ExternalObjectDirectoryEntity::getCreatedDateTime))
-            .orElseThrow(() -> new DartsApiException(FAILED_TO_DOWNLOAD_TRANSCRIPT));
-
         auditApi.recordAudit(DOWNLOAD_TRANSCRIPTION, userAccountEntity, transcriptionEntity.getCourtCase());
 
         return DownloadTranscriptResponse.builder()
-            .resource(getResourceStreamFor(latestExternalObjectDirectory))
+            .resource(getResourceStreamFor(latestTranscriptionDocument))
             .contentType(latestTranscriptionDocument.getFileType())
             .fileName(latestTranscriptionDocument.getFileName())
-            .externalLocation(latestExternalObjectDirectory.getExternalLocation())
             .transcriptionDocumentId(latestTranscriptionDocument.getId()).build();
     }
 
     private UserAccountEntity getUserAccount() {
         return userIdentity.getUserAccount();
     }
 
-    private InputStreamResource getResourceStreamFor(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
-        final UUID externalLocation = externalObjectDirectoryEntity.getExternalLocation();
-        InputStreamResource stream;
-        if (externalObjectDirectoryEntity.isForLocationType(UNSTRUCTURED)) {
-            stream = new InputStreamResource(dataManagementApi.getBlobDataFromUnstructuredContainer(externalLocation).toStream());
-        } else if (externalObjectDirectoryEntity.isForLocationType(INBOUND)) {
-            stream = new InputStreamResource(dataManagementApi.getBlobDataFromInboundContainer(externalLocation).toStream());
-        } else {
+    private InputStreamResource getResourceStreamFor(TranscriptionDocumentEntity latestTranscriptionDocument) {
+        DatastoreContainerType containerTypeUsedToDownload = null;
+        try {
+            DownloadResponseMetaData downloadResponseMetaData = storageBlobFileDownloadHelper.getDownloadResponse(
+                latestTranscriptionDocument.getExternalObjectDirectoryEntities());
+            containerTypeUsedToDownload = downloadResponseMetaData.getContainerTypeUsedToDownload();
+            if (!downloadResponseMetaData.isSuccessfulDownload()) {
+                throw new DartsApiException(FAILED_TO_DOWNLOAD_TRANSCRIPT);
+            }
+            return new InputStreamResource(downloadResponseMetaData.getInputStream());
+        } catch (IOException e) {
+            log.error("Failed to download transcript file using latestTranscriptionDocument ID {}, containerTypeUsedToDownload = {}",
+                      latestTranscriptionDocument.getId(),
+                      containerTypeUsedToDownload,
```

Comment:
will containerTypeUsedToDownload contain anything if the exception is thrown from downloadResponseMetaData?

Reply:
I'd expect it to be `null`

Reply:
so do we need it in the log message?

Reply:
Done in commit aa6e83cd39b9126692767e4e9999507dfedd94d2
* Update DownloadableExternalObjectDirectories#getFileBasedDownload to remove throws IOException